src/components/LibraryManageDialog.vue

```diff
@@ -0,0 +1,603 @@
+            voiceSamplePaths: styleInfo.voiceSamples,
+          } as StyleInfo;
+        }),
+        policy: speaker.speakerInfo.policy,
+      },
+    };
+  });
+};
+
+watch(
+  modelValueComputed,
+  async (newValue) => {
+    if (!newValue) {
+      return;
+    }
+    const engineIdsWithNotFetchingDownloadableLibraries = [];
+    for (const engineId of engineIdsWithDownloadableLibraries.value) {
+      if (
+        fetchStatuses.value[engineId] === "fetching" ||
+        fetchStatuses.value[engineId] === "success"
+      ) {
+        continue;
+      }
+      fetchStatuses.value[engineId] = "fetching";
+      engineIdsWithNotFetchingDownloadableLibraries.push(engineId);
+    }
+    for (const engineId of engineIdsWithNotFetchingDownloadableLibraries) {
+      const fetchResult = await store
+        .dispatch("INSTANTIATE_ENGINE_CONNECTOR", {
+          engineId,
+        })
+        .then((instance) =>
+          Promise.all([
+            instance.invoke("downloadableLibrariesDownloadableLibrariesGet")(
+              {}
+            ),
+            instance.invoke("installedLibrariesInstalledLibrariesGet")({}),
+          ])
+        )
+        .then((libraryEndpoints): [
+          BrandedDownloadableLibrary[],
+          BrandedInstalledLibrary[]
+        ] => {
+          fetchStatuses.value[engineId] = "success";
+          return [
+            libraryEndpoints[0].map((library) => {
+              return {
+                ...library,
+                uuid: LibraryId(library.uuid),
+                speakers: libraryInfoToCharacterInfos(engineId, library),
+              };
+            }),
+            Object.entries(libraryEndpoints[1]).map(([uuid, library]) => {
+              return {
+                ...library,
+                uuid: LibraryId(uuid),
+                speakers: libraryInfoToCharacterInfos(engineId, library),
+              };
+            }),
+          ];
+        })
+        .catch((e) => {
+          fetchStatuses.value[engineId] = "error";
+          store.dispatch("LOG_ERROR", e);
+        });
+
+      if (!fetchResult) return;
+
+      const [brandedDownloadableLibraries, brandedInstalledLibraries] =
+        fetchResult;
+      downloadableLibraries.value[engineId] = brandedDownloadableLibraries;
+      installedLibraries.value[engineId] = brandedInstalledLibraries;
+
+      const libraries = downloadableLibraries.value[engineId] || [];
+
+      libraries.sort((a, b) => {
+        const toPrimaryOrder = (library: BrandedDownloadableLibrary) => {
+          const localLibrary = installedLibraries.value[engineId].find(
+            (l) => l.uuid === library.uuid
+          );
+          // アップデート > 未インストール > インストール済み の順
+          if (!localLibrary) {
+            return 1;
+          } else if (semver.gt(library.version, localLibrary?.version)) {
+            return 2;
+          } else {
+            return 0;
+          }
+        };
+        return toPrimaryOrder(b) - toPrimaryOrder(a);
+      });
```

Comment:
```suggestion

        const toPrimaryOrder = (library: BrandedDownloadableLibrary) => {
          const localLibrary = installedLibraries.value[engineId].find(
            (l) => l.uuid === library.uuid
          );
          // アップデート > 未インストール > インストール済み の順
          if (!localLibrary) {
            return 1;
          } else if (semver.gt(library.version, localLibrary?.version)) {
            return 2;
          } else {
            return 0;
          }
        };
        libraries.sort((a, b) => {
          return toPrimaryOrder(b) - toPrimaryOrder(a);
        });
```
（自分が書いたコードですが）
toPrimaryOrderは外側で定義出来ると思います。

Reply:
そうですね、たしかに内部に入れる必要はなさそうです。

まあでも引数が複雑化しそうなので、提案してもらった階層のままでいいかなと...!(ここ以外で使う予定はないと思うので、汎用性を上げる必要はないかなと思いました)